service/matching/tasklist/matcher.go

```diff
@@ -501,6 +528,9 @@ func (tm *TaskMatcher) pollOrForward(
 			EventName:    "Poll Timeout",
 		})
 		return nil, ErrNoTasks
+	case <-tm.cancelCtx.Done():
```

Comment:
we can chain the contexts including `cancelCtx` before forwarding polls at line 
`tm.fwdr.ForwardPoll(ctx)`

Reply:
Did chaining trick with `ctxutil.WithPropagatedContextCancel`.
I think now the semantics `we-dont-care-if-matcher-closes-or-client-disconnects` is more explicit.